blueocean-dashboard/src/main/js/PipelineRoutes.jsx

```diff
@@ -82,9 +83,16 @@ function persistModalBackground() {
  * the background context.
  */
 function handleNavigationChangeToFromModal(prevState, nextState, replace, callback) {
-    if (nextState.params.runId && (prevState == null || !prevState.params.runId)) {
+    const enterRunDetails = nextState.params.runId && (prevState == null || !prevState.params.runId);
+    const leaveRunDetails = (prevState !== null && prevState.params.runId) && !nextState.params.runId;
+    const enterCreatePipeline = nextState.location.pathname.indexOf('/create-pipeline') !== -1 &&
```

Comment:
This branch was forked off the one from #567 so this code has been cleaned up there (and your comments on that PR will be addressed)